src/twisted/trial/_dist/test/test_workerreporter.py

```diff
@@ -88,6 +90,30 @@ def test_addFailureTuple(self):
                          managercommands.AddFailure)
 
 
+    def test_addFailureNonASCII(self):
+        """
+        L{WorkerReporter.addFailure} sends a L{managercommands.AddFailure}
+        message when called with a L{Failure}, even if it includes encoded
+        non-ASCII content.
+        """
+        content = u"\N{SNOWMAN}".encode("utf-8")
+        exception = RuntimeError(content)
+        failure = Failure(exception)
+        self.workerReporter.addFailure(self.test, failure)
+        self.assertEqual(
+            self.fakeAMProtocol.lastCall,
+            managercommands.AddFailure,
+        )
+        self.assertEqual(
+            content,
+            self.fakeAMProtocol.lastArgs["fail"],
+        )
+    if _PY3:
+        test_addFailureNonASCII.skip = (
+            "Exceptions only convert to unicode on Python 3"
```

Comment:
From the implementation, I see there is no different behaviour.

I think that this should not be skipped on Py3 but rather update the implementation to have an exception on py3.

```
twisted.trial.unittest.FailTest: b'\xe2\x98\x83' != b"b'\\xe2\\x98\\x83'"
```

Reply:
Regarding the implementation having no different behavior, this is true but there is a difference in Python itself which is that str(Exception()) returns bytes on Python 2 and unicode on Python 3.

I'm not sure what you mean about updating the implementation to have an exception on Python 3 means.